From 9bc2be61817363a4ae46d2c621586e9e962b7fb8 Mon Sep 17 00:00:00 2001
From: Ziyuan Xu <xzy.xu@rock-chips.com>
Date: Sat, 2 Jun 2018 14:50:14 +0800
Subject: [PATCH] rtc: hym8563: fix the incorrect initialization

The **valid** is always false in __probe__ function, so that the time
will be invalid at the first time.

Fixes: 66af9943e353 ("rtc: hym8563: add initialization data")
Change-Id: Id7a245159ab0fc4e9980db5e538b05ff17a02d89
Signed-off-by: Ziyuan Xu <xzy.xu@rock-chips.com>
---
 drivers/rtc/rtc-hym8563.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/rtc/rtc-hym8563.c b/drivers/rtc/rtc-hym8563.c
index ea9f4aabffa6..864f645e5e79 100644
--- a/drivers/rtc/rtc-hym8563.c
+++ b/drivers/rtc/rtc-hym8563.c
@@ -548,6 +548,7 @@ static int hym8563_probe(struct i2c_client *client,
 		return -ENOMEM;
 
 	hym8563->client = client;
+	hym8563->valid = true;
 	i2c_set_clientdata(client, hym8563);
 
 	device_set_wakeup_capable(&client->dev, true);
-- 
2.35.3

